examples/v1beta1/taskruns/step-timeout.yaml

```diff
@@ -0,0 +1,14 @@
+kind: TaskRun
+apiVersion: tekton.dev/v1beta1
+metadata:
+  generateName: timeout-test-
+spec:
+  taskSpec:
+    steps:
+      - name: wait
+        image: ubuntu
+        script: | 
+          #!/usr/bin/env bash
+          echo "I am supposed to sleep for 60 seconds!"
+          sleep 60
+        timeout: 5s
```

Comment:
examples can't currently fail b/c this will cause the integration tests to fail - also the examples are (currently) meant to be examples of successful tasks + pipelines (might change via https://github.com/tektoncd/pipeline/issues/3076 !)

so this would need to be an end to end test instead